src/Functions/generateSerialID.cpp

```diff
@@ -0,0 +1,188 @@
+#include <Common/ZooKeeper/ZooKeeper.h>
+#include <Common/ZooKeeper/KeeperException.h>
+#include <Common/escapeForFileName.h>
+#include <Columns/ColumnsNumber.h>
+#include <Columns/ColumnString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Interpreters/Context.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+#include <Core/ServerSettings.h>
+#include <Core/Settings.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int LIMIT_EXCEEDED;
+    extern const int BAD_ARGUMENTS;
+}
+
+namespace ServerSetting
+{
+    extern const ServerSettingsString series_keeper_path;
+}
+
+namespace Setting
+{
+    extern const SettingsUInt64 max_autoincrement_series;
+}
+
+
+namespace
+{
+
+class FunctionSerial : public IFunction
+{
+private:
+    ContextPtr context;
+    String keeper_path;
+
+public:
+    static constexpr auto name = "generateSerialID";
+
+    explicit FunctionSerial(ContextPtr context_) : context(context_)
+    {
+        keeper_path = context->getServerSettings()[ServerSetting::series_keeper_path];
+        zkutil::ZooKeeperPtr keeper = context->getZooKeeper();
+        size_t max_series = context->getSettingsRef()[Setting::max_autoincrement_series];
+
+        Coordination::Stat stat;
+        if (keeper->exists(keeper_path, &stat))
+        {
+            if (static_cast<size_t>(stat.numChildren) >= max_series)
```

Comment:
A small observation: currently it will prevent even using already created series.